lib/puppet/pops/pcore.rb

```diff
@@ -47,6 +47,10 @@ def self.init_env(loader)
             # Task metadata
             metadata => { type => Hash[String, Any] },
 
+            # The path of the module the task came from
+            # TODO should we try to make this a regex?
+            module => { type => Optional[String] },
```

Comment:
Why would this be a regex?

Reply:
Also, it's always going to be present, so I wouldn't make it optional.

Reply:
because module names are restricted. The answer is no but that will be clearer if we rename this attr to "module_dir" or something